src/SamplesServiceModelGenerator.Tests/Swagger/ParserTests.cs

```diff
@@ -0,0 +1,141 @@
+﻿using FluentAssertions;
+using Newtonsoft.Json.Linq;
+using NUnit.Framework;
+using SamplesServiceModelGenerator.Swagger;
+using Property = SamplesServiceModelGenerator.Swagger.Property;
+
+namespace SamplesServiceModelGenerator.Tests.Swagger
+{
+    [TestFixture]
+    public class ParserTests
+    {
+        const string _baseUrl = "https://demo.aqsamples.com/api/swagger.json";
+        private string _jsonText = "";
+
+        private Parser _testparser = new Parser();
+        private Api _parseOutput = new Api();
+        private JObject _json = new JObject();
+
+        private string LoadStringFromUrl(string url)
+        {
+            var uri = new Uri(url);
+
+            Console.WriteLine($"Fetching {uri} ...");
+
+            using (var client = new HttpClient())
+            {
+                var response = client.GetAsync(uri).Result;
+                return response.Content.ReadAsStringAsync().Result;
+            }
+        }
+
+        [OneTimeSetUp]
+        public void ForEachTest()
+        {
+            _jsonText = LoadStringFromUrl(_baseUrl);
+
+            try
+            {
+                _json = JObject.Parse(_jsonText);
+            }
+            catch (Exception ex)
+            {
+                throw new Exception("Test Swagger is not a Valid Json", ex);
+            }
+            _parseOutput = _testparser.Parse(_jsonText, _baseUrl);
+        }
+
+        [Test]
+        public void ParseBaseUrl()
+        {
+            Assert.AreEqual(_baseUrl, _parseOutput.BaseUrl);
+        }
+
+        [Test]
+        public void ParseTitle()
```

Comment:
```suggestion
        public void Parse_SamplesSwaggerDoc_ParsesTitle()
```

And so on for the remaining tests. Alternatively, you can lump all of these into one `[Test]`, and create separate assertion methods like:
```
[Test]
public void Parse_SamplesSwaggerDoc_Succeeds() {
  AssertExpectedBaseUrl();
  AssertExpectedTitle();
  // the other test assertions
  ...
  ...
}

private void AssertExpectedBaseUrl() {
  Assert.AreEqual(_parseOutput.BaseUrl, _baseUrl);
}

// add other assertion methods
````

This way, you can also have an invalid test case like:
```
[Test]
public void Parse_InvalidSamplesSwaggerDoc_Throws() {
  var invalidJson = "{ baseUrl: invalidUrl, title: fakeTitle }"  // Basically any invalid json object
  var action = () => _testParser.Parse(invalidJson, _baseUrl);

  Assert.Throws(action);
}
```

Reply:
Thanks it's a great suggestion. I incorporated multiple assertions in one test and added the invalid case too :)